index.js

```diff
@@ -95,8 +95,11 @@ module.exports = function(timeout, filter) {
      */
     getHandler: function () {
       var sequence = Array.prototype.slice.call(arguments);
-      return function(files, done) {
-        enqueue.apply(null, sequence.concat(done));
+      return function (files, done) {
+        if (typeof done === 'function' || typeof done === 'string') {
```

Comment:
Please check the branch, I just updated it with the rationale for each
test.  I understand the concern of "what's in sequence, and what can I do
with it?" is handled elsewhere there, so I tried to be as liberal as
possible with ``done`` while touching your code as little as possible. :)

On Sun, Jan 10, 2016 at 11:06 PM, Ben Holloway <notifications@github.com>
wrote:

> In index.js
> <https://github.com/bholloway/gulp-watch-sequence/pull/2#discussion_r49285162>
> :
>
> > @@ -95,8 +95,11 @@ module.exports = function(timeout, filter) {
> >       */
> >      getHandler: function () {
> >        var sequence = Array.prototype.slice.call(arguments);
> > -      return function(files, done) {
> > -        enqueue.apply(null, sequence.concat(done));
> > +      return function (files, done) {
> > +        if (typeof done === 'function' || typeof done === 'string') {
>
> @nuba <https://github.com/nuba> From what I can see of the latest
> gulp-watch it only ever calls with a single file argument
> <https://github.com/floatdrop/gulp-watch/blob/master/index.js#L139>.
>
> The two arguments I have are a hangover from a previous version (1.0.7)
> where done may be a function as a result of using gulp-batch
> <https://github.com/floatdrop/gulp-watch/blob/v1.0.7/index.js#L37>. So
> your detection of function allows both versions of gulp-watch to be used.
> I think it should be included (as you have done) for correctness but I am
> not sure it serves a practical purpose.
>
> However I do not see how done could be a string value. Is this just
> defensive programming?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bholloway/gulp-watch-sequence/pull/2/files#r49285162>.
>